src/parse/matches/arg_matches.rs

```diff
@@ -234,14 +234,31 @@ impl ArgMatches {
             fn to_str_slice(o: &OsString) -> &str {
                 o.to_str().expect(INVALID_UTF8)
             }
-            let to_str_slice: fn(&OsString) -> &str = to_str_slice; // coerce to fn pointer
 
             Values {
-                iter: arg.vals.iter().map(to_str_slice),
+                iter: arg.vals_flatten().map(to_str_slice),
             }
         })
     }
 
+    /// Placeholder documentation.
+    pub fn grouped_values_of<T: Key>(&self, id: T) -> Option<GroupedValues> {
+        #[allow(clippy::type_complexity)]
+        let arg_values: for<'a> fn(
+            &'a MatchedArg,
+        ) -> Map<
+            std::slice::Iter<'a, Vec<OsString>>,
+            fn(&Vec<OsString>) -> Vec<&str>,
+        > = |arg| {
+            arg.vals()
+                .map(|g| g.iter().map(|x| x.to_str().expect(INVALID_UTF8)).collect())
+        };
+        self.args
+            .get(&Id::from(id))
+            .map(arg_values)
```

Comment:
Put the lambda from above directly here.

Reply:
That's makes rustc complaining about type imcompatibility(To make type compatible, we need two level of Closure -> fn pointer coercion, but that's not currently implemented in rustc).